src/main/java/ch/njol/skript/expressions/ExprSets.java

```diff
@@ -63,6 +63,13 @@ public class ExprSets extends SimpleExpression<Object> {
 				"[(all [[of] the]|the|every)] item(s|[ ]types)", "[(all [[of] the]|the)] items of type[s] %itemtypes%",
 				"[(all [[of] the]|the|every)] block(s|[ ]types)", "[(all [[of] the]|the)] blocks of type[s] %itemtypes%",
 				"([all [[of] the]] colo[u]rs|(the|every) colo[u]r)");
+
+		ExprLoopValue.registerLoopValueHandler(ExprSets.class, (source, type) -> {
+			int pattern = source.pattern;
+			return (pattern == 4 && (type.equalsIgnoreCase("color") || type.equalsIgnoreCase("colour")))
+			|| (pattern >= 2 && type.equalsIgnoreCase("block"))
+			|| (pattern < 2 && type.equalsIgnoreCase("item"));
```

Comment:
```suggestion
			return (pattern == 4 && (type.equalsIgnoreCase("color") || type.equalsIgnoreCase("colour")))
					|| (pattern >= 2 && type.equalsIgnoreCase("block"))
					|| (pattern < 2 && type.equalsIgnoreCase("item"));
```
or 1 tab less, I don't remember which we do

Reply:
Also, can you check if these are needed? Cause I don't think they are, assuming they expression has the right return type.

Reply:
The `loop all blocks` actually returns an ItemStack - `loop all items` and `loop all blocks` is the same thing

Although right now the get method is broken and `loop all blocks` causes an NPE

Reply:
I've decided to keep only "block" since `loop all blocks` returns an itemstack